library/utils.py

```diff
@@ -52,3 +54,22 @@ def format_url(path: str, subpath: str) -> str:
         return url.replace("s3://", "/vsis3/")
 
     return url
+
+def get_execution_details():
+    today = date.today().strftime("%Y-%m-%d")
```

Comment:
Went with date rather than timestamp just because the timestamp itself is a little arbitrary depending on when in the process it's actually taken. But if we think it'd be more useful to have slightly more specific info, would be happy to use timestamp instead

Reply:
agree it can be arbitrary, but seems worth the timestamp to avoid conflicts/confusion if we run something multiple times a day (like when debugging)

hour:minute would probably be precise enough and still readable